src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -311,4 +331,52 @@ mod test {
             assert_eq!(got, exp);
         }
     }
+
+    #[test]
+    fn test_year() {
+        let tests = vec![
+            ("0000-00-00 00:00:00", -1i64),
+            ("1-01-01 01:01:01", 1i64),
+            ("2018-01-01 01:01:01", 2018i64),
+            ("2019-01-01 01:01:01", 2019i64),
+            ("2020-01-01 01:01:01", 2020i64),
+            ("2021-01-01 01:01:01", 2021i64),
+            ("2022-01-01 01:01:01", 2022i64),
+            ("2023-01-01 01:01:01", 2023i64),
+            ("2024-01-01 01:01:01", 2024i64),
+            ("2025-01-01 01:01:01", 2025i64),
+            ("2026-01-01 01:01:01", 2026i64),
+            ("2027-01-01 01:01:01", 2027i64),
+            ("2028-01-01 01:01:01", 2028i64),
+            ("2029-01-01 01:01:01", 2029i64),
+        ];
+        let mut ctx = EvalContext::default();
+        for (arg, exp) in tests {
+            let arg = datum_expr(Datum::Time(Time::parse_utc_datetime(arg, 6).unwrap()));
+            let exp = Datum::I64(exp);
+            let f = scalar_func_expr(ScalarFuncSig::Year, &[arg]);
+            let op = Expression::build(&mut ctx, f).unwrap();
+            let got = op.eval(&mut ctx, &[]).unwrap();
+            assert_eq!(got, exp);
+        }
+        // test NULL case
+        let input = datum_expr(Datum::Null);
+        let f = scalar_func_expr(ScalarFuncSig::Year, &[input]);
+        let op = Expression::build(&mut ctx, f).unwrap();
+        let got = op.eval(&mut ctx, &[]).unwrap();
+        assert_eq!(got, Datum::Null);
+
+        // test zero case
+        let cfg = EvalConfig::new()
```

Comment:
Oh, `EvalConfig` is updated, you need to do it like this now: https://github.com/tikv/tikv/pull/3622/files#diff-1377468ba7a8bb8132fa4d0c4f429700L281